torch/lib/THPP/generators/THCGenerator.cpp

```diff
@@ -0,0 +1,40 @@
+#include "THCGenerator.hpp"
+
+#define const_generator_cast(generator) \
+  dynamic_cast<const THCGenerator&>(generator)
+
+namespace thpp {
+
+THCGenerator::THCGenerator(THCState* state))
+  : generator(THCRandom_getGenerator())
```

Comment:
`THCRandom_getGenerator()` is not declared in any `.h` file, so this won't compile either - moreover, the `THCRandom_getGenerator` from `THC` takes a `state` argument